File: worker/photoflow/keys.py
"""Every object key the system uses, in one place.

The gallery declares the same keys in src/storage/keys.ts, so changing one here
means changing that too.
"""

INCOMING = "incoming/"
CATALOG_MANIFEST = "catalog/manifest.json"
CATALOG_SHARDS = "catalog/shards/"
CATALOG_EMBEDDINGS = "catalog/embeddings/"
META_STATE = "meta/state.json"
META_LOGS = "meta/log/"
META_HEARTBEAT = "meta/heartbeat.json"
META_REPROCESS = "meta/reprocess/"


def original(prefix: str, file_id: str) -> str:
    return f"original/{prefix}{file_id}"


def tile(prefix: str, file_id: str) -> str:
    return f"tile-image/{prefix}{file_id}.jpeg"


def preview(prefix: str, file_id: str, extension: str) -> str:
    return f"preview/{prefix}{file_id}{extension}"


def reprocess_request(file_id: str) -> str:
    return f"{META_REPROCESS}{file_id}.json"


def shard(month: str, part: int = 1) -> str:
    """A month's shard, or one part of it when the month is too big for one file.

    The first part keeps the plain name, so a catalog written before months could
    be split still reads correctly.
    """
    suffix = "" if part <= 1 else f".p{part}"
    return f"{CATALOG_SHARDS}{month}{suffix}.json"


def embeddings(month: str) -> str:
    return f"{CATALOG_EMBEDDINGS}{month}.bin"

File: src/storage/keys.ts
// The object keys the worker writes and this app reads. Mirrors
// worker/photoflow/keys.py — changing one means changing both.

export const INCOMING = 'incoming/';
export const CATALOG_MANIFEST = 'catalog/manifest.json';
export const META_STATE = 'meta/state.json';
export const META_LOGS = 'meta/log/';
export const META_HEARTBEAT = 'meta/heartbeat.json';
export const META_REPROCESS = 'meta/reprocess/';

// A month too large for one file is split into parts. The first keeps the plain
// name, so a catalog written before splitting existed still reads correctly.
// Media keys. The worker builds the same strings in keys.py; the app used to be
// handed them as URL prefixes in the manifest, which is no longer written because
// every URL is now signed here from the key.
export const original = (fileId: string) => `original/${fileId}`;
export const tile = (fileId: string) => `tile-image/${fileId}.jpeg`;
export const preview = (fileId: string, extension: string) => `preview/${fileId}${extension}`;

export const shard = (month: string, part = 1) =>
    `catalog/shards/${month}${part > 1 ? `.p${part}` : ''}.json`;
export const embeddings = (month: string) => `catalog/embeddings/${month}.bin`;
export const deviceLog = (deviceId: string) => `${META_LOGS}${deviceId}.json`;
export const reprocessRequest = (fileId: string) => `${META_REPROCESS}${fileId}.json`;
